common-custom-user-data-maven-extension/src/main/java/com/gradle/GroovyScriptUserData.java

```diff
@@ -28,20 +34,29 @@ private static File getScriptFile(MavenSession session) {
         return new File(rootDir, ".mvn/gradle-enterprise-custom-user-data.groovy");
     }
 
-    private static void evaluateGroovyScript(MavenSession session, BuildScanApi buildScan, Logger logger, File scriptFile) throws MavenExecutionException {
+    private static void evaluateGroovyScript(MavenSession session, @Nullable BuildScanApi buildScan, @Nullable BuildCacheApi buildCache, Logger logger, File scriptFile) throws MavenExecutionException {
         try {
-            Binding binding = prepareBinding(session, buildScan, logger);
+            Binding binding = prepareBinding(session, buildScan, buildCache, logger);
             new GroovyShell(GroovyScriptUserData.class.getClassLoader(), binding).evaluate(scriptFile);
         } catch (IOException e) {
             throw new MavenExecutionException("Failed to evaluate custom user data Groovy script: " + scriptFile, e);
         }
     }
 
-    private static Binding prepareBinding(MavenSession session, BuildScanApi buildScan, Logger logger) {
+    private static Binding prepareBinding(MavenSession session, @Nullable BuildScanApi buildScan, @Nullable BuildCacheApi buildCache, Logger logger) {
         Binding binding = new Binding();
         binding.setVariable("project", session.getTopLevelProject());
         binding.setVariable("session", session);
-        binding.setVariable("buildScan", buildScan);
+        if (buildScan != null) {
+            binding.setVariable("buildScan", buildScan);
+        } else {
+            logger.debug("Cannot bind buildScan API.");
```

Comment:
I wonder whether this should be a warning to make users aware they are using an outdated version of the Gradle Enterprise Maven Extension.

Reply:
You mean below, right? Do we get into these `else` blocks at all? If so, in which cases?

Reply:
When the `BuildCacheApi` component couldn't be found (https://github.com/gradle/gradle-enterprise-build-config-samples/pull/37/files#diff-252fd27e9805a7ae34557c0a8426795eR48) This can happen when a user uses an old Maven extension version that doesn't have the `BuildCacheApi` class I guess.

Reply:
But then the we'd get a `ClassNotFoundError` for `BuildCacheApi`, wouldn't we?

Reply:
Yes I think so as well. Should we remove these logs then?

Reply:
I think so.

Reply:
Removed

Reply:
@marcphilipp @guylabs Since you both 'think so' but do not know for sure, we need to ensure what the actual behavior is. @guylabs can you please check it? Thanks.

Reply:
It will never evaluate the Groovy script if one of the API's are `null`. See https://github.com/gradle/gradle-enterprise-build-config-samples/blob/79cdd2aeb9f988f7d653196d6493033363c20d22/common-custom-user-data-maven-extension/src/main/java/com/gradle/GroovyScriptUserData.java#L18-L30 So all good.

Reply:
No, that is not waht the code is saying.